internal/xds/bootstrap/util.go

```diff
@@ -6,19 +6,45 @@
 package bootstrap
 
 import (
+	"fmt"
+
+	bootstrapv3 "github.com/envoyproxy/go-control-plane/envoy/config/bootstrap/v3"
+
 	egv1a1 "github.com/envoyproxy/gateway/api/v1alpha1"
-	yamlutils "github.com/envoyproxy/gateway/internal/utils/yaml"
+	"github.com/envoyproxy/gateway/internal/utils/proto"
+	_ "github.com/envoyproxy/gateway/internal/xds/extensions" // DON'T REMOVE: import of all extensions
 )
 
 // ApplyBootstrapConfig applies the bootstrap config to the default bootstrap config and return the result config.
 func ApplyBootstrapConfig(boostrapConfig *egv1a1.ProxyBootstrap, defaultBootstrap string) (string, error) {
 	bootstrapType := boostrapConfig.Type
 	if bootstrapType != nil && *bootstrapType == egv1a1.BootstrapTypeMerge {
-		mergedBootstrap, err := yamlutils.MergeYAML(defaultBootstrap, boostrapConfig.Value)
+		mergedBootstrap, err := mergeBootstrap(defaultBootstrap, boostrapConfig.Value)
```

Comment:
can you highlight the limitation of the existing function ?

Reply:
the current code handles the unmarshal and validation later in https://github.com/envoyproxy/gateway/blob/ef1934f1461157afc890cddac863418c3e90b4f0/api/v1alpha1/validation/envoyproxy_validate.go#L155

Reply:
can you take a look at the linked issue?

Reply:
sounds like we panicked in `yaml.mergeMaps`, why not fix that ?

Reply:
IMO, the best choice is let envoy do that using `---config-path`, this's logic from kuma project is more likely same as envoy did.

Also, I don't want to spend endless type handling in `yaml.mergeMaps`, if someone know that better me, I'm ok to close this.